alma_client/request.py

```diff
@@ -58,22 +48,36 @@ def expectPaginatedList(self, cls, next_page):
         self.response_processor = lambda response: PaginatedResults(response.json, cls, next_page)
         return self
 
-    @configure_credentials
     def get(self):
         self.method = "GET"
         return self
 
-    @configure_credentials
     def post(self):
         self.method = "POST"
         return self
 
-    @configure_credentials
     def put(self):
         self.method = "PUT"
         return self
 
-    @configure_credentials
     def delete(self):
         self.method = "DELETE"
         return self
+
+    @property
+    def data(self):
+        return json.dumps(self.body)
+
+    def to_httpx(self):
+        self.context.credentials.configure(self)
+        headers = dict(self.headers)
```

Comment:
Intent is more explicit with a copy
```suggestion
        headers = self.headers.copy()
```